packages/core/database/src/query/helpers/populate/apply.ts

```diff
@@ -529,24 +565,24 @@ const morphToOne = async (input, ctx) => {
   // make a map for each type what ids to return
   // make a nested map per id
 
-  const idsByType = results.reduce((acc, result) => {
-    const idValue = result[morphColumn.idColumn.name];
-    const typeValue = result[morphColumn.typeColumn.name];
+  const idsByType = results.reduce<Record<string, ID[]>>((acc, result) => {
+    const idValue = result[morphColumn.idColumn.name] as ID;
+    const typeValue = result[morphColumn.typeColumn.name] as string;
 
     if (!idValue || !typeValue) {
       return acc;
     }
 
-    if (!_.has(typeValue, acc)) {
-      acc[typeValue] = [];
+    if (!(typeValue in acc)) {
+      acc[typeValue] = [idValue];
```

Comment:
With the change, we're adding the idValue and then also pushing it below. Is that intended?

Reply:
nope nice catch